### PR TITLE
chore(driver): fix integration test retry configuration

### PR DESCRIPTION
**Pull request by emilyrohrbough**
Test flakiness in `packages/driver` has been observed and tests have not been retried as expected. This change enables the retry configuration in the integration tests via the cypress configuration.

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
Looking at the failures / flakey tests listed in the last Cypress comment, retries have been enabled.

File: packages/driver/cypress/support/defaults.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 const { $ } = Cypress
 
-let isActuallyInteractive
+const isActuallyInteractive = Cypress.config('isInteractive')
```
Comment (emilyrohrbough) [inline packages/driver/cypress/support/defaults.js:3]: This is the change to enable retries

File: packages/driver/cypress/support/utils.js [hunk 2]
```diff
@@ -66,6 +66,12 @@ export const shouldBeCalledWithCount = (num) => (stub) => wrapped(stub).should('
 
 export const shouldNotBeCalled = (stub) => wrapped(stub).should('not.be.called')
 
+export const assertLogLength = (logs, expectedLength) => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/support/utils.js:69]: Helper to log the list of logs received to help debug when log length is different than expected.

File: packages/driver/cypress/integration/commands/navigation_spec.js [hunk 3]
```diff
@@ -85,7 +85,7 @@ describe('src/cy/commands/navigation', () => {
     })
 
     // TODO: fix this
-    it.skip('(FLAKY) sets timeout to Cypress.config(pageLoadTimeout)', {
+    it('(FLAKY) sets timeout to Cypress.config(pageLoadTimeout)', {
```
Comment (mjhenkes) [inline packages/driver/cypress/integration/commands/navigation_spec.js:88]: With retries setup, are these passing reliably now?
```suggestion
    it('sets timeout to Cypress.config(pageLoadTimeout)', {
```

File: packages/driver/cypress/integration/commands/navigation_spec.js [hunk 4]
```diff
@@ -303,7 +303,7 @@ describe('src/cy/commands/navigation', () => {
...
```
Comment (mjhenkes) [inline packages/driver/cypress/integration/commands/navigation_spec.js:306]:
```suggestion
    it('sets timeout to Cypress.config(pageLoadTimeout)', {
```

File: packages/driver/cypress/integration/commands/navigation_spec.js [hunk 5]
```diff
@@ -546,7 +546,7 @@ describe('src/cy/commands/navigation', () => {
 
   context('#visit', () => {
     // TODO: fix this
-    it.skip('(FLAKY) sets timeout to Cypress.config(pageLoadTimeout)', {
+    it('(FLAKY) sets timeout to Cypress.config(pageLoadTimeout)', {
```
Comment (mjhenkes) [inline packages/driver/cypress/integration/commands/navigation_spec.js:549]:
```suggestion
    it('sets timeout to Cypress.config(pageLoadTimeout)', {
```

**Review by flotwig — CHANGES_REQUESTED** [review]
My supplied test case from earlier works, but some stuff is still not retrying: https://app.circleci.com/pipelines/github/cypress-io/cypress/26345/workflows/825f5705-1c96-4ff5-b0d4-f43234e78612/jobs/994369/parallel-runs/1

There's no evidence of retrying on that failure in the logs.

You can observe this locally running `yarn cypress:run --headed --no-exit`, *some tests* will fail and not be retried, and *some* will be retried. Not sure what's happening, there's no per-test config in this file...

**Comment by flotwig** [comment]
> @flotwig You linked to the CircleCI logs. If you look in the [Cypress Dashboard results ](https://dashboard.cypress.io/projects/ypt4pf/runs/23408/test-results/93790e21-34ce-4d74-ad3a-929acb822437?statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D) you will see retries are working.

Some tests are retrying, some tests aren't retrying. That test code does work now, but some real specs in `assertions_spec` are still not retrying.

> > You can observe this locally running yarn cypress:run --headed --no-exit, some tests will fail and not be retried, and some will be retried. Not sure what's happening, there's no per-test config in this file...
> 
> I am currently running `yarn workspace @packages/driver cypress:run --headed` and have yet to observed this behavior , but there are a lot of tests here & haven't made it through the `cypress/integration/commands/actions` yet. It'll take while to get through them locally. Can you share which specs (or the video recording) where you observed this so I can dig in deeper?

Observe the results of `yarn cypress:run --headed --no-exit --spec **/assertions_spec.js` (to run the spec with the test that's not retrying in CI):

https://user-images.githubusercontent.com/1151760/140983890-7814d2b3-da51-4a47-b32c-f3d8740b6a9b.mp4

Earlier tests in the file do retry, so I'm not sure what causes this yet. But, it's not retrying.

**Comment by emilyrohrbough** [comment]
@flotwig  Thanks for the example. I'll take a look.

**Review by flotwig — APPROVED** [review]

File: packages/driver/src/cypress/mocha.ts [hunk 6]
```diff
@@ -416,6 +416,13 @@ const patchRunnableResetTimeout = () => {
     }
 
     this.timer = setTimeout(() => {
+      if (runnable.state === 'passed') {
+        // this timeout can be reached at the same time that a
+        // user does an asynchronous `done`, so double-check
+        // that the test has not already passed before timing out
+        return
+      }
```
Comment (emilyrohrbough) [inline packages/driver/src/cypress/mocha.ts:424]: For anyone curious to reproduce this issue, you can comment this out and runt his test to consistently see the error shown in the video Zach linked: 

```js
      it('should not be true', { defaultCommandTimeout: 1 }, (done) => {
        cy.on('fail', (err) => {
          expect(err.message).to.eq('expected false to be true')

          done()
        })

        cy.noop(false).should('be.true')
      })
```

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**